### DISCUSSION TITLE
CodePipeline + Github status checks

### DISCUSSION THREAD

**Discussion by abr-storm** [aws/aws-cdk#30032] [Q&A]
Hi guys,

I'm trying to work out how to update Github status checks based on the pass/fail of a CodePipeline.

My pipeline sources a Github repo with a call to `CodePipelineSource.connection` with a CodeStar connection arn instead of `CodePipelineSource.github` with an access token, as advised by the docs.

The pipeline triggers in response to repo changes as I would expect, but Github is not updated with the success/failure after the pipeline runs.

Has anyone got status checks working with this kind of setup? How did you do it? The little green tick is very important to me

**Comment by NelsonLopez69** [comment]
I think this post may help https://paul-hands-phd.medium.com/using-aws-codebuild-to-set-up-github-continuous-integration-19b92efbd094

**Comment by ticoombs** [comment]
Out of the box it isn't supported like it is via CodeBuild. 

I finally found: https://repost.aws/questions/QUBTn5ZT8kQTqmk-65ydpy4Q/how-do-you-integrate-a-codestar-connection-with-github-status-checks

If you want to use CodePipeline as the initiator, then you need to do something like: https://aws.amazon.com/es/blogs/devops/aws-codepipeline-build-status-in-a-third-party-git-repository/ (which uses https://github.com/aws-samples/aws-blog-codepipeline-third-party-git-build-status/tree/main) 

Otherwise, you can use the link NelsonLopez referenced. Which uses "CodeBuild" as the initiator, instead of "CodePipeline". 
Then instead of completely refactoring, you'd end up having to do something like: Github -> Codebuild -> S3 -> which triggers -> CodePipeline

**Reply by nickwillan** [reply]
A really easy option is to use something like [Pipechecks.io](https://www.pipechecks.io).
